ompi/mca/osc/rdma/osc_rdma_dynamic.c

```diff
@@ -428,10 +428,6 @@ static int ompi_osc_rdma_refresh_dynamic_region (ompi_osc_rdma_module_t *module,
         }
         peer->regions = temp;
 
-        /* lock the region */
-        ompi_osc_rdma_lock_acquire_shared (module, &peer->super, 1, offsetof (ompi_osc_rdma_state_t, regions_lock),
```

Comment:
Can you say a word why you removed the lock? AFAICS, the owner of the region takes an exclusive lock in `ompi_osc_rdma_attach` and the shared lock here is needed to ensure the consistency of the region information.

Reply:
I'm seeing a hang in this code path. From what I can see, because this process already has an exclusive lock on this peer, it can't acquire the shared lock (it thinks someone else owns it), and will loop for infinity trying to acquire this lock. To me this looks like a double-lock scenerio. @hjelmn can probably say if that is intended or not, perhaps there is a different fix needed for this case.